server/resolvers/TestPlanRunOperations/updateResultResolver.js

```diff
@@ -0,0 +1,40 @@
+const {
+    getTestPlanRunById,
+    updateTestPlanRun
+} = require('../../models/services/TestPlanRunService');
+const populatedDataResolver = require('../PopulatedData');
+
+// this would come from the iframe result after submitting a test
+const updateResultResolver = async (
+    { parentContext: { id: testPlanRunId } },
+    { result, index }
+) => {
+    const testPlanRun = await getTestPlanRunById(testPlanRunId);
+    let { testResults } = testPlanRun;
+
+    const testResultToUpdateIndex = testResults.findIndex(
+        testResult => testResult.test.executionOrder === index
+    );
+
+    if (!result) {
+        // if user decides to start over
+        testResults[testResultToUpdateIndex].result = null;
+        testResults[testResultToUpdateIndex].test.isComplete = false;
+        testResults[testResultToUpdateIndex].test.completedAt = null;
+    } else {
+        testResults[testResultToUpdateIndex].result = {
+            ...testResults[testResultToUpdateIndex].result,
+            ...result
+        };
+        testResults[testResultToUpdateIndex].test.isComplete = true;
+        testResults[testResultToUpdateIndex].test.completedAt = new Date();
+    }
```

Comment:
Using a TestResultInput type might help result in a lot of this code not being needed - let me know when you're ready to work on it and maybe we can pair on it!

Reply:
Sure, we could do that